src/test/java/org/wise/portal/dao/work/impl/HibernateEventDaoTest.java

```diff
@@ -0,0 +1,362 @@
+/**
+ * Copyright (c) 2006 Encore Research Group, University of Toronto
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.wise.portal.dao.work.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.wise.portal.dao.work.EventDao;
+import org.wise.portal.domain.authentication.Gender;
+import org.wise.portal.domain.authentication.Schoollevel;
+import org.wise.portal.domain.group.Group;
+import org.wise.portal.domain.run.Run;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.workgroup.Workgroup;
+import org.wise.portal.junit.AbstractTransactionalDbTests;
+import org.wise.vle.domain.work.Event;
+
+/**
+ * @author Geoffrey Kwan
+ */
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class HibernateEventDaoTest extends AbstractTransactionalDbTests {
+  
+  private Group period1, period2;
+  private User teacher1, student1, student2;
+  private Run run;
+  private Workgroup workgroup1, workgroup2;
+  private List<JSONObject> components;
+  private Event event1;
+  
+  @Autowired
+  private EventDao eventDao;
```

Comment:
```suggestion
  private EventDao<Event> eventDao;
```
This will remove the remaining 26 warnings.